File: src/refactoring-exercise/code-replacer/dot-net/RefactoringExample/RefactoringExample/CodeReplacerTest.cs

using System;
using System.IO;
using NUnit.Framework;

namespace RefactoringExample
{
    [TestFixture]
    public class CodeReplacerTest
    {
        CodeReplacer replacer;

        [SetUp]
        public void SetUp()
        {
            replacer = new CodeReplacer();
        }

        [Test]
        public void TestTemplateLoadedProperly()
        {
            try
            {
                replacer.Substitute("anything", new StreamWriter(Stream.Null));
            }
            catch (Exception e)
            {
                Assert.Fail("No exception expected, but saw:" + e);
            }
            Assert.AreEqual("xxx%CODE%yyy%ALTCODE%zzz\n", replacer.SourceTemplate);
        }

        [Test]
        public void TestSubstitution()
        {
            StringWriter stringOut = new StringWriter();
            String trackingId = "01234567";

            try
            {
                replacer.Substitute(trackingId, stringOut);
            }
            catch (IOException ex)
            {
                Assert.Fail("testSubstitution exception - " + ex);
            }
            catch (Exception e)
            {
            }

            Assert.AreEqual("xxx01234567yyy0123-4567zzz\n", stringOut.ToString());
        }
    }
}

File: src/refactoring-exercise/code-replacer/java/src/test/CodeReplacerTest.java
import java.io.*;
import junit.framework.*;

public class CodeReplacerTest extends TestCase {
    CodeReplacer replacer;

    public CodeReplacerTest(String testName) {super(testName);}

    protected void setUp() {replacer = new CodeReplacer();}

    public void testTemplateLoadedProperly() {
        try {
            replacer.substitute("anything", new PrintWriter(new StringWriter()));
        } catch (Exception ex) {
            fail("No exception expected, but saw:" + ex);
        }

        assertEquals(
            "xxx%CODE%yyy%ALTCODE%zzz\n",
            replacer.sourceTemplate);
    }

    public void testSubstitution() {
        StringWriter stringOut = new StringWriter();
        PrintWriter testOut = new PrintWriter (stringOut);
        String trackingId = "01234567";
        
        try {
            replacer.substitute(trackingId, testOut);
        } catch (IOException ex) {
            fail ("testSubstitution exception - " + ex);
        }

        assertEquals("xxx01234567yyy01234-567zzz\n", stringOut.toString());
    }

}